src/arch/cuda/hybridAPI/hapi_impl.cpp

```diff
@@ -249,43 +252,78 @@ void GPUManager::init() {
 // which depends on the compute capability of the device.
 // Returns the number of created streams.
 int GPUManager::createStreams() {
-  if (streams_)
-    return n_streams_;
+  int new_n_streams = 0;
 
-#if CMK_SMP || CMK_MULTICORE
   if (device_prop_.major == 3) {
     if (device_prop_.minor == 0)
-      n_streams_ = 16;
+      new_n_streams = 16;
     else if (device_prop_.minor == 2)
-      n_streams_ = 4;
+      new_n_streams = 4;
     else // 3.5, 3.7 or unknown 3.x
-      n_streams_ = 32;
+      new_n_streams = 32;
   }
   else if (device_prop_.major == 5) {
     if (device_prop_.minor == 3)
-      n_streams_ = 16;
+      new_n_streams = 16;
     else // 5.0, 5.2 or unknown 5.x
-      n_streams_ = 32;
+      new_n_streams = 32;
   }
   else if (device_prop_.major == 6) {
     if (device_prop_.minor == 1)
-      n_streams_ = 32;
+      new_n_streams = 32;
     else if (device_prop_.minor == 2)
-      n_streams_ = 16;
+      new_n_streams = 16;
     else // 6.0 or unknown 6.x
-      n_streams_ = 128;
+      new_n_streams = 128;
   }
   else // unknown (future) compute capability
-    n_streams_ = 128;
-#else
-  n_streams_ = 4; // FIXME per PE in non-SMP mode
+    new_n_streams = 128;
+#if !CMK_SMP && !CMK_MULTICORE
+  // Allocate total physical streams between GPU managers sharing a device...
+  // i.e. PEs / num devices
+  int device_count;
+  hapiCheck(cudaGetDeviceCount(&device_count));
+  int pes_per_device = CmiNumPesOnPhysicalNode(0) / device_count;
+  pes_per_device = pes_per_device > 0 ? pes_per_device : 1;
+  new_n_streams =  (new_n_streams + pes_per_device - 1) / pes_per_device;
```

Comment:
This splits the pool of streams to different in PEs in non-SMP mode, but what's the expected behavior for SMP mode if the user has more than one process per GPU?